packages/snaps-rpc-methods/src/endowments/cronjob.ts

```diff
@@ -62,7 +62,10 @@ export const cronjobEndowmentBuilder = Object.freeze({
  */
 export function getCronjobCaveatMapper(
   value: Json,
-): Pick<PermissionConstraint, 'caveats'> {
+): Pick<PermissionConstraint, 'caveats'> | Record<string, never> {
+  if (Object.keys(value as Record<string, Json>).length === 0) {
```

Comment:
```suggestion
  if (!value || !isObject(value) || Object.keys(value).length === 0) {
```
Instead of casting I think it's preferable to use the same pattern as we use in other mappers